kernels/test/op_grid_sampler_2d_test.cpp

```diff
@@ -0,0 +1,374 @@
+        7.0,
+        8.0,
+        9.0,
+        10.0,
+        11.0,
+        12.0,
+        13.0,
+        14.0,
+        15.0,
+        16.0};
+    Tensor input = tf.make(input_sizes, input_data);
+
+    // Create a 1x2x2x2 grid tensor with points outside the input boundaries
+    const std::vector<int32_t> grid_sizes = {1, 2, 2, 2};
+    std::vector<float> grid_data = {
+        -2.0,
+        -2.0, // Way outside top-left
+        2.0,
+        -2.0, // Way outside top-right
+        -2.0,
+        2.0, // Way outside bottom-left
+        2.0,
+        2.0 // Way outside bottom-right
+    };
+    Tensor grid = tf.make(grid_sizes, grid_data);
+
+    // Create output tensor with expected shape 1x1x2x2
+    const std::vector<int32_t> output_sizes = {1, 1, 2, 2};
+    Tensor out = tf.zeros(output_sizes);
+
+    // Set testing modes
+    const int64_t interpolation_mode = 0;
+    const int64_t padding_mode = 1; // border
+    const bool align_corners = false;
+
+    torch::executor::aten::grid_sampler_2d_outf(
+        context_,
+        input,
+        grid,
+        interpolation_mode,
+        padding_mode,
+        align_corners,
+        out);
+
+    // Expected output
+    const std::vector<float> expected_output = {
+        1.0,
+        4.0, // Top corners
+        13.0,
+        16.0 // Bottom corners
+    };
+
+    // Test
+    EXPECT_TENSOR_CLOSE(out, tf.make(output_sizes, expected_output));
+  }
+
+  // Test grid_sampler_2d with reflection padding mode
+  void test_grid_sampler_2d_reflection_padding() {
+    TensorFactory<ScalarType::Float> tf;
+
+    // Create a 1x1x4x4 input tensor
+    const std::vector<int32_t> input_sizes = {1, 1, 4, 4};
+    std::vector<float> input_data = {
+        1.0,
+        2.0,
+        3.0,
+        4.0,
+        5.0,
+        6.0,
+        7.0,
+        8.0,
+        9.0,
+        10.0,
+        11.0,
+        12.0,
+        13.0,
+        14.0,
+        15.0,
+        16.0};
+    Tensor input = tf.make(input_sizes, input_data);
+
+    // Create a 1x2x2x2 grid with points outside the input boundaries
+    const std::vector<int32_t> grid_sizes = {1, 2, 2, 2};
+    std::vector<float> grid_data = {
+        -2.0,
+        -2.0, // Way outside top-left
+        2.0,
+        -2.0, // Way outside top-right
+        -2.0,
+        2.0, // Way outside bottom-left
+        2.0,
+        2.0 // Way outside bottom-right
+    };
+    Tensor grid = tf.make(grid_sizes, grid_data);
+
+    // Create output tensor with expected shape 1x1x2x2
+    const std::vector<int32_t> output_sizes = {1, 1, 2, 2};
+    Tensor out = tf.zeros(output_sizes);
+
+    // Set testing modes
+    const int64_t interpolation_mode = 0;
+    const int64_t padding_mode = 2; // reflection
+    const bool align_corners = false;
+
+    torch::executor::aten::grid_sampler_2d_outf(
+        context_,
+        input,
+        grid,
+        interpolation_mode,
+        padding_mode,
+        align_corners,
+        out);
+
+    // Check non-zero
```

Comment:
this comment just repeats the code, delete (and similar comments throughout that repeat the code)